src/modules/ekf2/ekf2_main.cpp

```diff
@@ -670,11 +666,12 @@ bool Ekf2::init()
 	// if EKF2_IMU_ID is non-zero we use the corresponding IMU, otherwise the voted primary (sensor_combined)
 	if (device_id != 0) {
 		for (int i = 0; i < MAX_SENSOR_COUNT; i++) {
-			vehicle_imu_s imu{};
+			_vehicle_imu_sub.ChangeInstance(i);
```

Comment:
This should check the return value (or make it succeed in all cases, see other comment).

Reply:
Done.